pkgs/by-name/ar/archipelago/package.nix

```diff
@@ -1,43 +1,62 @@
 {
   lib,
+  pkgs,
+  stdenvNoCC,
   appimageTools,
+  makeWrapper,
+  xsel,
+  xclip,
+  mtdev,
+  zenity,
   fetchurl,
   nix-update-script,
-  extraPackages ? [ ],
 }:
-let
+stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "archipelago";
   version = "0.6.3";
+
   src = fetchurl {
-    url = "https://github.com/ArchipelagoMW/Archipelago/releases/download/${version}/Archipelago_${version}_linux-x86_64.AppImage";
+    url = "https://github.com/ArchipelagoMW/Archipelago/releases/download/${finalAttrs.version}/Archipelago_${finalAttrs.version}_linux-x86_64.AppImage";
     hash = "sha256-PetlGYsdhyvThIFqy+7wbPLAXDcgN2Kcl2WF3rta8PA=";
   };
 
-  appimageContents = appimageTools.extractType2 { inherit pname version src; };
-in
-appimageTools.wrapType2 {
-  inherit pname version src;
-  extraPkgs =
-    pkgs:
-    [
-      pkgs.xsel
-      pkgs.xclip
-      pkgs.mtdev
-    ]
-    ++ extraPackages;
-  extraInstallCommands = ''
-    install -Dm444 ${appimageContents}/archipelago.desktop -t $out/share/applications
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  runtimeDependencies = [
+    xsel
+    xclip
+    mtdev
+  ]
+  ++ appimageTools.defaultFhsEnvArgs.targetPkgs pkgs
+  ++ appimageTools.defaultFhsEnvArgs.multiPkgs pkgs;
+
+  appimageContents = appimageTools.extractType2 { inherit (finalAttrs) pname version src; };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    makeWrapper ${finalAttrs.appimageContents}/AppRun $out/bin/archipelago \
+      --set APPDIR '${finalAttrs.appimageContents}' \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath finalAttrs.runtimeDependencies} \
```

Comment:
What about autoPatchelfHook instead of setting an LD_LIBRARY_PATH?

Reply:
Thanks for the suggestion, uncovered some extra dependencies and now it runs without `nix-ld`! It seems `mtdev` still needs to be in `LD_LIBRARY_PATH` though, just being in `runtimeDependencies` doesn't seem to be enough.
```
OSError: libmtdev.so.1: cannot open shared object file: No such file or directory
```

Due to that I went ahead and left the derivation's original dependencies in `LD_LIBRARY_PATH`.

Reply:
> You might be able to avoid the LD_LIBRARY_PATH by manually adding an rpath for the package libmtdev.so.1's from, I've seen this used a few times.
> 
> ```
> postFixup = ''
>     patchelf --add-rpath ${lib.makeLibraryPath [ mtdev ]} $out/bin/mybin
> '';
> ```

Unfortunately it doesn't seem we can do that in this case since it's loaded from a compiled `.pyc` file instead of a regular ELF or shared library. However I did realize that `xsel` and `xclip` should probably be in `binPath` since kivy uses `subprocess.Popen` to run them so I've moved them there. I think `mtdev` is the only thing that needs to be on `LD_LIBRARY_PATH`.

Let me know if you have any other ideas.